homeassistant/components/obihai/sensor.py

```diff
@@ -46,16 +46,26 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     pyobihai = PyObihai()
 
+    login = pyobihai.check_account(host, username, password)
+    if not login:
+        _LOGGER.error("Invalid credentials")
+        return
+
     services = pyobihai.get_state(host, username, password)
```

Comment:
It might make sense to store the credentials on the client instance, so the library user doesn't have to pass them into every method call.

It's just a suggestion for the future.

Reply:
Always appreciate these suggestions! Will get this done in the next PR when I add `unique_id` and `availability`.